fml/message_loop_impl.cc

```diff
@@ -119,17 +119,22 @@ void MessageLoopImpl::DoTerminate() {
 
 void MessageLoopImpl::FlushTasks(FlushType type) {
   TRACE_EVENT0("fml", "MessageLoop::FlushTasks");
-  std::vector<fml::closure> invocations;
 
-  task_queue_->GetTasksToRunNow(queue_id_, type, invocations);
-
-  for (const auto& invocation : invocations) {
+  const auto now = fml::TimePoint::Now();
+  for (;;) {
```

Comment:
is there a way to change this to `While(TasksAvailableOnQueue(queue_id_))`?

Reply:
Yeah, we can create a new method to just return a bool if task is available, and another method to return the actual task. Although I think that might be redundant. 
What about make `GetNextTaskToRun` return a bool, and also takes a pointer so the invocation is set as well?
For example `bool GetNextTaskToRun(TaskQueueId queue_id, fml:closure& invocation)` ?

Reply:
+1

Reply:
done